Sources/StreamChat/Models/ChatMessage.swift

```diff
@@ -413,7 +413,12 @@ public extension ChatMessage {
 
 extension ChatMessage: Hashable {
     public static func == (lhs: Self, rhs: Self) -> Bool {
-        lhs.id == rhs.id
+        lhs.id == rhs.id &&
```

Comment:
How did you come up with these properties? Maybe we can think about unifying it with this one: https://github.com/GetStream/stream-chat-swift/blob/081601c1e9cb925e8aa0f6470f8e3120add5ca54/Sources/StreamChatUI/ChatMessageList/ChatMessageListView%2BDiffKit.swift#L34.

Reply:
Some trial and error but I realised now that I need to add a lot more due to the offline support where the updatedAt won't get changed if I am offline. Sounds like it makes sense to copy what we do for UIKit list updates.

Reply:
I unified them with what we have for DiffKit.

Reply:
do you think it might have performance impact?

Reply:
Need to profile but since we compare more properties then some impact indeed. Don't think it is going to be much :)

Reply:
Profiled it a bit and seems like the longer == implementation triggers Swift to copy the passed in values.
<img width="943" alt="Screenshot 2024-07-04 at 13 37 40" src="https://github.com/GetStream/stream-chat-swift/assets/1469907/e6ac29f0-4175-45a0-b0e2-e0d1d612bf45">
OR
the shorter == implementation is inlined and I never see it in the profiler. I think it is OK, in the future we probably want to go for making these types non-copiable anyway.